examples/simple_with_session.rs

```diff
@@ -0,0 +1,79 @@
+//! Simple Goose load test example. Duplicates the simple example on the
+//! Locust project page (https://locust.io/).
+//!
+//! ## License
+//!
+//! Copyright 2020 Jeremy Andrews
+//!
+//! Licensed under the Apache License, Version 2.0 (the "License");
+//! you may not use this file except in compliance with the License.
+//! You may obtain a copy of the License at
+//!
+//! http://www.apache.org/licenses/LICENSE-2.0
+//!
+//! Unless required by applicable law or agreed to in writing, software
+//! distributed under the License is distributed on an "AS IS" BASIS,
+//! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//! See the License for the specific language governing permissions and
+//! limitations under the License.
+
+use goose::prelude::*;
+use serde::Deserialize;
+
+struct Session {
+    jwt_token: String,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AuthenticationResponse {
+    jwt_token: String,
+}
+
+fn main() -> Result<(), GooseError> {
+    GooseAttack::initialize()?
+        // In this example, we only create a single taskset, named "WebsiteUser".
+        .register_taskset(
+            taskset!("WebsiteUser")
+                // After each task runs, sleep randomly from 5 to 15 seconds.
+                .set_wait_time(5, 15)?
+                // This task only runs one time when the user first starts.
+                .register_task(task!(website_signup).set_on_start())
+                // These next two tasks run repeatedly as long as the load test is running.
+                .register_task(task!(authenticated_index)),
+        )
+        .execute()?
+        .print();
+
+    Ok(())
+}
+
+/// Demonstrates how to log in and set a session when a user starts. We flag this task as an
+/// on_start task when registering it above. This means it only runs one time
+/// per user, when the user thread first starts.
+async fn website_signup(user: &mut GooseUser) -> GooseTaskResult {
+    let request_builder = user.goose_post("/signup")?;
+    // https://docs.rs/reqwest/*/reqwest/blocking/struct.RequestBuilder.html#method.form
+    let params = [("username", "test_user"), ("password", "")];
+    let response = user
+        .goose_send(request_builder.form(&params), None)
+        .await?
+        .response?
+        .json::<AuthenticationResponse>()
+        .await?;
+
+    user.set_session_data(Session {
+        jwt_token: response.jwt_token,
+    });
+
+    Ok(())
+}
+
+/// A very simple task that simply loads the front page.
+async fn authenticated_index(user: &mut GooseUser) -> GooseTaskResult {
+    let session = user.get_session_data_uncheck::<Session>();
```

Comment:
My initial testing of this example led to a panic. I feel it would be better to provide an example using something like:
```rust
    if let Some(session) = user.get_session_data::<Session>() {
```

...instead of the `_uncheck()` version, under the assumption that people will copy these examples when writing their own load tests.  In the event that `get_session_data` fails, I imagine as we would expect session data we should set a failure: https://docs.rs/goose/0.13.3/goose/goose/struct.GooseUser.html#method.set_failure

Reply:
(Or another option: add a comment about why this might panic, and about instead using `get_session_data`... if the goal is just as simple of code as possible.)

Reply:
I added a comment because `set_failure`  require a `GooseRequestMetric` which is not available at that moment. 

Maybe we could enhance `GooseTaskError` to handle custom error.